pkg/compact/compact.go

```diff
@@ -221,23 +227,42 @@ func (s *Syncer) GarbageCollect(ctx context.Context) error {
 
 	begin := time.Now()
 
-	garbageIds := s.duplicateBlocksFilter.DuplicateIDs()
-	for _, id := range garbageIds {
+	duplicateIDs := s.duplicateBlocksFilter.DuplicateIDs()
+	deletionMarkMap := s.ignoreDeletionMarkFilter.DeletionMarkBlocks()
+
+	// GarbageIDs contains the duplicateIDs, since these blocks can be replaced with other blocks.
+	// We also remove ids present in deletionMarkMap since these blocks are already marked for deletion.
+	garbageIDs := []ulid.ULID{}
+	for _, id := range duplicateIDs {
```

Comment:
I thought you moved to map so we can simplify this loop... right?

Reply:
🤦‍♀ I thought this couldn't be done. This is done now